bundle/src/main/java/com/adobe/acs/commons/analysis/jcrchecksum/impl/options/RequestChecksumGeneratorOptions.java

```diff
@@ -93,13 +94,14 @@ public RequestChecksumGeneratorOptions(SlingHttpServletRequest request) throws I
         return paths;
     }
 
-    private static Set<String> getPathsFromInputstream(InputStream is) throws IOException {
+    private static Set<String> getPathsFromInputstream(InputStream is, String encoding) throws IOException {
         if (is == null) {
             return Collections.EMPTY_SET;
         }
 
         Set<String> paths = new HashSet<String>();
-        BufferedReader br = new BufferedReader(new InputStreamReader(is));
+        encoding = (encoding != null) ?  encoding : Charset.defaultCharset().name();
```

Comment:
I would prefer to do this check before passing "ëncoding" to this method. Caller should pass either default or other value.